udata/forms/fields.py

```diff
@@ -180,6 +180,19 @@ def __init__(self, *args, **kwargs):
         self.stacked = kwargs.pop('stacked', False)
         super(BooleanField, self).__init__(*args, **kwargs)
 
+    def process_formdata(self, valuelist):
+        if not valuelist:
+            return 
+
+        if valuelist[0] is None:
+            self.data = self.default
+            return
+
+        if valuelist[0] in self.false_values:
+            self.data = False
+        else:
+            self.data = True
```

Comment:
Should we raise an error if valuelist[0] is not a bool value?

Reply:
Fallback to the official implementation so it's not our problem :-)